src/cmd/ksh93/sh/jobs.c

```diff
@@ -1209,6 +1209,61 @@ int job_kill(register struct process *pw,register int sig)
 }
 
 /*
+ * Similar to job_kill, but dedicated to SIGHUP handling when session is
+ * being disconnected.
+ */
+int
+job_hup(struct process *pw, int sig)
+{
+	struct process	*px;
+	pid_t	pid;
+	int	r;
+
+	if (pw->p_pgrp == 0 || (pw->p_flag & P_DISOWN))
```

Comment:
I am curious how ```SIGHUP``` will be handled for processes in current process group if we return from here.